### PR TITLE
Feature to support setting twin reported properties during simulation

### PR DESCRIPTION
**Pull request by avagraw**
---
This project has adopted the [Microsoft Open Source Code of Conduct](https://opensource.microsoft.com/codeofconduct/). For more information see the [Code of Conduct FAQ](https://opensource.microsoft.com/codeofconduct/faq/) or contact [opencode@microsoft.com](mailto:opencode@microsoft.com) with any additional questions or comments.

Thank you for contributing to the IoT extension!

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] If introducing new functionality or modified behavior, are they backed by unit and integration tests?
- [ ] In the same context as above are command names and their parameter definitions accurate? Do help docs have sufficient content?
- [ ] Have **all** unit **and** integration tests passed locally? i.e. `pytest <project root> -vv`
- [ ] Have static checks passed using the .pylintrc and .flake8 rules? Look at the CI scripts for example usage.
- [ ] Have you made an entry in HISTORY.rst which concisely explains your feature or change?

### CODE REVIEW

**Comment by avagraw** [comment]
Integration tests pass:

image

**Comment by avagraw** [comment]
Integration tests pass, except a couple:

image

Failed integration tests also pass:

image
image
image

**Review by c-ryan-k — APPROVED** [review]

File: azext_iot/operations/_mqtt.py [hunk 1]
```diff
@@ -105,8 +109,19 @@ def twin_patch_handler(self, patch):
...
+
         try:
+            if self.init_reported_properties:
+                twin_properties = {
+                    "properties": {
+                        "desired": self.init_reported_properties
+                    }
+                }
+
+                _iot_device_twin_update(self.target, self.device_id, twin_properties)
```
Comment (digimaun) [inline azext_iot/operations/_mqtt.py:123]: The goal of this functionality is to be able to init device side (reported) properties. The implementation here is setting service side desired properties.

Comment (avagraw) [reply azext_iot/operations/_mqtt.py:123]: Update to the desired properties is picked up by the MQTT listener, which automatically updates the reported properties, hence bringing the desired and reported properties in sync.

Comment (digimaun) [reply azext_iot/operations/_mqtt.py:123]: But you don't **need** to set desired properties to apply the initial reported state. Devices connecting to IoT Hub will provide their initial state through reported properties without needing to make a modification to desired. Desired twin properties are a solution side interaction and its strange that we would force an update to the desired twin state in order to update the reported side when using facilities of a device simulation command. The primary justification that I see right now is implementation convenience.

You can take a [look at this device sample](https://github.com/Azure/azure-iot-sdk-python/blob/master/azure-iot-device/samples/pnp/temp_controller_with_thermostats.py) as an example/datapoint - the device connects to iothub initializing a set of reported properties (reflecting a pnp device) with no modification to the desired/solution side interaction.

**Pull request merged**